### DISCUSSION TITLE
Are there ways to parse strings instead of files?

### DISCUSSION THREAD

**Discussion by Teajey** [swc-project/swc#6365] [Q&A]
I want to be able to programmatically format some Typescript that I've generated in my project https://github.com/Teajey/accord

I feel like I should be able to input an unformatted string of Typescript, and receive a formatted string of Typescript as output. But so far I've only been able to find functions, such as `parse_js`, in these libraries that take a path to a file as input. That would be fine except for in my tests I don't want to have to write Typescript to a file in order to format it...

So is this possible, or am I barking up the wrong tree?

**Comment by hyf0** [comment]
You should be able to use `swc_ecma_parser` directly, instead of `swc` crate.
See https://github.com/swc-project/swc/blob/main/crates/swc_ecma_parser/examples/typescript.rs

**Reply by Viijay-Kr** [reply]
@Teajey Were you able to format the typescript output ? If yes can you please share your approach. I m working on a similar project and I would expect the output TS source should be formatted

**Reply by Teajey** [reply]
Sorry @Viijay-Kr. It's been long enough that I actually can't exactly remember why I was trying to achieve this. I don't remember ever figuring out how to programmatically format Typescript strings

**Reply by Viijay-Kr** [reply]
All good . Thanks for the reply. I m thinking of using Biome formatter for my use case

**Reply by acao** [reply]
@Viijay-Kr if you prefer prettier's formatting, you can just use the prettier [API libraries](https://prettier.io/docs/en/api#prettierformatsource-options)!
